support/scripts/multiboot2.py

```diff
@@ -0,0 +1,245 @@
+
+            new_phdr[0:ELF64_PHDR_LEN] = phdr[0:ELF64_PHDR_LEN]  # keep the rest
+            new_phdr[8:16] = self.prpnd_off(phdr[8:16], mb2_header_end_off)  # p_offset
+
+            return new_phdr
+
+        return [update_phdr(p) for p in self.elf64_phdrs]
+
+    def get_shdrs(self, mb2_header_end_off):
+        def update_shdr(shdr):
+            new_shdr = bytearray(ELF64_SHDR_LEN)
+
+            new_shdr[0:ELF64_SHDR_LEN] = shdr[0:ELF64_SHDR_LEN]  # keep the rest
+            new_shdr[24:32] = self.prpnd_off(
+                shdr[24:32], mb2_header_end_off
+            )  # sh_offset
+
+            return new_shdr
+
+        return [update_shdr(s) for s in self.elf64_shdrs]
+
+
+def add_tags(tags_opt):
+    tags_hdr = b""
+    for tag_type in tags_opt:
+        if (
+            tag_type == MULTIBOOT_HEADER_TAG_END
+            or tag_type == MULTIBOOT_HEADER_TAG_MODULE_ALIGN
+        ):
+            size = 8
+            tags_hdr += (
+                tag_type.to_bytes(2, "little")
+                + MULTIBOOT_HEADER_TAG_OPTIONAL.to_bytes(2, "little")
+                + size.to_bytes(4, "little")
+            )
+        else:
+            raise ValueError(f"Unknown tag type: {tag_type}")
+    return tags_hdr
+
+
+def get_multiboot2_hdr():
+    mb2_magic = MULTIBOOT_HEADER_MAGIC
+    mb2_architecture = MULTIBOOT_ARCHITECTURE_I386
+
+    tags_opt = [MULTIBOOT_HEADER_TAG_MODULE_ALIGN, MULTIBOOT_HEADER_TAG_END]
+    tags_hdr = add_tags(tags_opt)
+    tags_size = len(tags_hdr)
+
+    mb2_header_length = MULTIBOOT_HEADER_BASE_SIZE + tags_size
+    # We need to convert checksum to 4-byte signed integer range to avoid overflow
+    mb2_checksum = -(mb2_magic + mb2_architecture + mb2_header_length)
+    mb2_checksum = (mb2_checksum + 2 ** 32) % 2 ** 32
+    if mb2_checksum >= 2 ** 31:
+        mb2_checksum -= 2 ** 32
+
+    return (
+        mb2_magic.to_bytes(4, "little")
+        + mb2_architecture.to_bytes(4, "little")
+        + mb2_header_length.to_bytes(4, "little")
+        + mb2_checksum.to_bytes(4, "little", signed=True)
+        + tags_hdr
+    )
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Appends the Multiboot2 header to the beginning of an ELF64 Binary."
+    )
+    parser.add_argument("elf64", help="path to ELF64 binary to update")
+    opt = parser.parse_args()
+
+    elf64 = Elf64Editor(opt.elf64)
+
+    mb2_hdr = get_multiboot2_hdr()
+    mb2_header_end_off = ELF64_EHDR_LEN + len(mb2_hdr)
+
+    elf64_ehdr = elf64.get_ehdr(mb2_header_end_off)
+    elf64_phdrs = elf64.get_phdrs(mb2_header_end_off)
+    elf64_shdrs = elf64.get_shdrs(mb2_header_end_off)
+
+    with open(opt.elf64, "r+b") as new_elf:
+        # Save original ELF content
+        elf64_buf = new_elf.read()
+
+        # Go back to the beginning
+        new_elf.seek(0)
+
+        # Write ELF64 header with updated offsets
+        new_elf.write(elf64_ehdr)
+
+        # Append Multiboot2 header
```

Comment:
> The Multiboot2 header must be contained completely within the first 32768 bytes of the OS image, and must be 64-bit aligned.

The same way `multiboot.py` checks for alignment
```python
        # Append the Multiboot header
        # NOTE: Multiboot header must be 4-byte aligned - luckily that is the
        # case with the size of the ELF32 header (40 bytes)
        if elf32.tell() & 0b11:
            raise Exception("ELF32 header is not a multiple of 4 bytes in size")
```
You should do the same here.